diff -rupN cm/services/camera/libcameraservice/api1/CameraClient.cpp av/services/camera/libcameraservice/api1/CameraClient.cpp
--- cm/services/camera/libcameraservice/api1/CameraClient.cpp	2013-11-28 17:25:59.000000000 +0100
+++ av/services/camera/libcameraservice/api1/CameraClient.cpp	2013-12-01 11:57:36.000000000 +0100
@@ -701,6 +701,7 @@ status_t CameraClient::sendCommand(int32
         enableMsgType(CAMERA_MSG_STATS_DATA);
     } else if (cmd == CAMERA_CMD_HISTOGRAM_OFF) {
         disableMsgType(CAMERA_MSG_STATS_DATA);
+#if 0
     } else if (cmd == CAMERA_CMD_METADATA_ON) {
         enableMsgType(CAMERA_MSG_META_DATA);
     } else if (cmd == CAMERA_CMD_METADATA_OFF) {
@@ -708,6 +709,7 @@ status_t CameraClient::sendCommand(int32
     } else if ( cmd == CAMERA_CMD_LONGSHOT_ON ) {
         mLongshotEnabled = true;
     } else if ( cmd == CAMERA_CMD_LONGSHOT_OFF ) {
+#endif
         mLongshotEnabled = false;
         disableMsgType(CAMERA_MSG_SHUTTER);
         disableMsgType(CAMERA_MSG_COMPRESSED_IMAGE);
